### PR TITLE
Rename MoneyEffect member variables to prevent shadowing

### PR DESCRIPTION
**Pull request by tupaschoal**
Continues #10653

### CODE REVIEW

File: src/openrct2/rct1/S4Importer.cpp [hunk 1]
```diff
@@ -1705,11 +1705,11 @@ class S4Importer final : public IParkImporter
 
     void ImportMoneyEffect(MoneyEffect* dst, MoneyEffect* src)
     {
-        dst->move_delay = src->move_delay;
-        dst->num_movements = src->num_movements;
-        dst->value = src->value;
-        dst->offset_x = src->offset_x;
-        dst->wiggle = src->wiggle;
+        dst->MoveDelay = src->MoveDelay;
```
Comment (duncanspumpkin) [inline src/openrct2/rct1/S4Importer.cpp:1708]: This isn't right. S4 should have a RCT12SpriteMoneyEffect. It's not a mistake with this pr though.

File: src/openrct2/world/MoneyEffect.cpp [hunk 2]
```diff
@@ -47,16 +47,16 @@ void MoneyEffect::CreateAt(money32 value, int32_t x, int32_t y, int32_t z, bool
     if (moneyEffect == nullptr)
         return;
 
-    moneyEffect->value = value;
-    moneyEffect->vertical = (vertical ? 1 : 0);
+    moneyEffect->Value = value;
+    moneyEffect->Vertical = (vertical ? 1 : 0);
```
Comment (duncanspumpkin) [inline src/openrct2/world/MoneyEffect.cpp:51]: I believe that vertical is actually whether it is a credit or debit. We should make this an enum class and define the two states.

File: src/openrct2/rct1/S4Importer.cpp (hunk 1 above)
Comment (IntelOrca) [reply src/openrct2/rct1/S4Importer.cpp:1708]: I don't think it is a mistake, just we haven't got round to changing it yet. They all used to use the same structs, we have gradually been copying them and removing the push, 1 pop.

Comment (duncanspumpkin) [reply src/openrct2/rct1/S4Importer.cpp:1708]: Yes but for now we don't enforce packing on money effect so who knows if this works atm.

File: src/openrct2/world/MoneyEffect.cpp (hunk 2 above)
Comment (tupaschoal) [reply src/openrct2/world/MoneyEffect.cpp]: I think it has two meanings, I did convert it to an enum, but the `GetStringId` makes things look a bit weird. I've added the commit for you guys to see

File: src/openrct2/world/Sprite.h [hunk 3]
```diff
@@ -74,14 +74,20 @@ struct Duck : SpriteGeneric
...
-    static void CreateAt(money32 value, int32_t x, int32_t y, int32_t z, bool vertical);
+    enum class TransactionType : bool
+    {
+        CREDIT,
+        DEBIT,
+    };
+
+    uint16_t MoveDelay;
+    uint8_t NumMovements;
+    TransactionType CashFlow;
```
Comment (Gymnasiast) [inline src/openrct2/world/Sprite.h:85]: I am not entirely convinced this is indeed credit/debit. Would have to check the calls. I am not sure about the name of this field either.

File: src/openrct2/world/MoneyEffect.cpp [hunk 4]
```diff
@@ -147,11 +147,12 @@ void MoneyEffect::Update()
 
 std::pair<rct_string_id, money32> MoneyEffect::GetStringId() const
 {
-    rct_string_id spentStringId = vertical ? STR_MONEY_EFFECT_SPEND_HIGHP : STR_MONEY_EFFECT_SPEND;
-    rct_string_id receiveStringId = vertical ? STR_MONEY_EFFECT_RECEIVE_HIGHP : STR_MONEY_EFFECT_RECEIVE;
+    rct_string_id spentStringId = CashFlow == TransactionType::CREDIT ? STR_MONEY_EFFECT_SPEND_HIGHP : STR_MONEY_EFFECT_SPEND;
```
Comment (Gymnasiast) [inline src/openrct2/world/MoneyEffect.cpp:150]:
```suggestion
    rct_string_id spentStringId = CashFlow == TransactionType::DEBIT ? STR_MONEY_EFFECT_SPEND_HIGHP : STR_MONEY_EFFECT_SPEND;
```

File: src/openrct2/world/MoneyEffect.cpp [hunk 5]
```diff
@@ -147,11 +147,12 @@ void MoneyEffect::Update()
...
+    rct_string_id receiveStringId = CashFlow == TransactionType::CREDIT ? STR_MONEY_EFFECT_RECEIVE_HIGHP
+                                                                        : STR_MONEY_EFFECT_RECEIVE;
```
Comment (Gymnasiast) [inline src/openrct2/world/MoneyEffect.cpp:152]: These lines show that `vertical` does not mean whether it credit/debit, IMO.

Comment (Gymnasiast) [reply src/openrct2/world/MoneyEffect.cpp:152]: I dug a bit further. The `HIGHP` strings contain currency strings with 2 decimals and are used for guest purchases. The only odd bit is why it increases Z for guest purchases.

Comment (duncanspumpkin) [reply src/openrct2/world/MoneyEffect.cpp:152]: Guest purchases were added later so they work differently. I'd forgotten about that. Renaming the variable might not make sense now.

**Comment by tupaschoal** [comment]
I've dropped the renaming of `Vertical`, as we agreed that it wasn't really just credit/debit. I think this is good to go then.

File: src/openrct2/rct1/S4Importer.cpp (hunk 1 above)
Comment (tupaschoal) [reply src/openrct2/rct1/S4Importer.cpp:1708]: I created #11757 to track this issue, feel free to add more details there.

**Pull request merged**